vignettes/irecist.Rmd

```diff
@@ -0,0 +1,858 @@
+[Creating ADRS (Including Non-standard Endpoints)](adrs.html). Most of the endpoints 
+are derived by calling `admiral::derive_extreme_event()`. Thus the examples in this 
+vignette can also be used as a starting point for implementing response criteria other 
+than iRECIST 1.1,  e.g., RECIST 1.1 or International Myeloma Working Group (IMWG) 
+criteria for the diagnosis of multiple myeloma.
+
+For more information on the iRECIST guidelines user may visit
+https://recist.eortc.org/recist-1-1-2/
+
+Examples are currently presented and tested using `ADSL` (ADaM) and
+`RS` (SDTM) inputs. However, other domains could be used. The functions 
+and workflow could similarly be used to create an intermediary `ADEVENT` ADaM.
+
+**Note**: *All examples assume CDISC SDTM and/or ADaM format as input
+unless otherwise specified.*
+
+# Programming Workflow
+
+-   [Read in Data](#readdata)
+-   [Pre-processing of Input Records](#input)
+-   [Derive Confirmed Progressive Disease Parameter](#pd)
+-   [Derive Response Parameter](#rsp)
+-   [Derive Clinical Benefit Parameter](#cb)
+-   [Derive Best Overall Response Parameter](#bor)
+-   [Derive Response Parameters requiring Confirmation](#confirm)
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of `ADRS` should be
+read into the environment. This will be a company specific process. Some
+of the data frames needed may be `ADSL`, `RS` and `TU`.
+
+For example purpose, the SDTM and ADaM datasets (based on CDISC Pilot
+test data)---which are included in `{pharmaversesdtm}` and `{pharmaverseadam}`---are used.
+
+```{r message=FALSE}
+library(admiral)
+library(admiralonco)
+library(dplyr)
+library(pharmaverseadam)
+library(pharmaversesdtm)
+library(lubridate)
+library(stringr)
+data("adsl")
+data("rs_onco_irecist")
+
+adsl_onco <- adsl
+rs <- rs_onco_irecist
+
+rs <- convert_blanks_to_na(rs)
+```
+
+```{r echo=FALSE}
+# select subjects from adsl such that there is one subject without RS data
+rs_subjects <- unique(rs$USUBJID)
+adsl_subjects <- unique(adsl_onco$USUBJID)
+adsl <- filter(
+  adsl_onco,
+  USUBJID %in% union(rs_subjects, setdiff(adsl_subjects, rs_subjects)[1])
+)
+```
+
+At this step, it may be useful to join `ADSL` to your `RS` domain. Only
+the `ADSL` variables used for derivations are selected at this step. The
+rest of the relevant `ADSL` would be added later.
+
+```{r eval=TRUE}
+adsl_vars <- exprs(RANDDT)
+adrs <- derive_vars_merged(
+  rs,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = exprs(STUDYID, USUBJID)
+)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, RSTESTCD, RSDTC, VISIT, RANDDT),
+  filter = RSTESTCD == "OVRLRESP"
+)
+```
+
+## Pre-processing of Input Records {#input}
+
+The first step involves company-specific pre-processing of records for
+the required input to the downstream parameter functions. It could involve
```

Comment:
In comparison to the adrs.html vignette, the sentence about reprocessing for BICR has been removed. I think it can be left in here.

Reply:
will add back